src/pages/SignupHandler/index.js

```diff
@@ -9,265 +12,295 @@ import {
-    };
-  }
+const mapStateToProps = state => {
+  const { auth } = state;
+  return auth;
+};
+
+const SignupHandler = props => {
+  const [errors, setErrors] = useState({
+    firstName: '',
+    lastName: '',
+    email: '',
+    passwordConstraints: '',
+    passwordConfirm: '',
+  });
+  const [constraints, setConstraints] = useState({
+    passwordLength: 'unmet',
+    passwordSpecialChars: 'unmet',
+    passwordContainsNumber: 'unmet',
+    passwordBlockLetter: 'unmet',
+  });
+
+  // Form fields
+  const [firstName, setFirstName] = useState('');
+  const [lastName, setLastName] = useState('');
+  const [email, setEmail] = useState('');
+  const [password, setPassword] = useState('');
+  const [confirmPassword, setConfirmPassword] = useState('');
+  const [enableSubmitBtn, setEnableSubmitButton] = useState(false);
 
-  componentDidMount = () => {
-    this.disableSubmitBtn();
+  /* eslint-disable no-restricted-syntax */
+  const validateForm = () => {
+    // return early.
+    if (
+      firstName === '' ||
+      lastName === '' ||
+      email === '' ||
+      password === '' ||
+      confirmPassword === ''
+    ) {
+      return false;
+    }
+    // check if no errors.
+    for (const dep of Object.entries(errors)) {
+      if (dep[1].length > 0) {
+        return false;
+      }
+    }
+    // check if all constraints are met.
+    for (const ct of Object.entries(constraints)) {
+      if (ct[1] !== 'met') {
+        return false;
+      }
+    }
+    // if we reach here, it means
+    // we have convered all of the edge cases.
+    return true;
   };
 
-  enableSubmitBtn = () => {
-    this.setState({
-      variantVal: 'primary',
-      btnColor: 'white',
-      btnDisabled: 'false',
-    });
+  useEffect(
+    () => {
+      if (validateForm()) {
+        setEnableSubmitButton(true);
+      }
+    },
+    [firstName, lastName, email, password, confirmPassword]
+  );
+
+  const getSplitDisplayProperty = (name, type) => {
+    if (constraints[name] === 'met' && type === 'check') {
+      return 'block';
+    }
+    if (constraints[name] === 'unmet' && type === 'close') {
+      return 'block';
+    }
+    return 'none';
   };
 
-  disableSubmitBtn = () => {
-    this.setState({
-      variantVal: 'tertiary',
-      btnColor: 'black',
-      btnDisabled: 'true',
-    });
+  const handleFirstNameInputChange = val => {
+    setFirstName(val);
   };
 
-  handleNameInputChange = username => {
-    const { email, password, confPassword } = this.state;
-    this.setState({
-      username,
-    });
-    if (username !== '' && email !== '' && password !== '' && confPassword !== '') {
-      this.enableSubmitBtn();
-    } else {
-      this.disableSubmitBtn();
-    }
+  const handleLastNameInputChange = val => {
+    setLastName(val);
   };
 
-  handleEmailInputChange = email => {
-    const { username, password, confPassword } = this.state;
-    this.setState({
-      email,
-    });
+  const handleEmailInputChange = val => {
```

Comment:
It seems there's an edge case in this section causing the registration form to proceed when inputing a correct email and then removing it. Can we debug this further? 

```
Steps to reproduce:
1. type "gkalsi@redhat.com" in email address input field
2. fill all other fields correctly
3. remove text from email address input field
```

Reply:
Whoa, Thanks Gurbir. It was in the back of my head but I forgot about it.
This should fix it.
```diff
     () => {
       if (validateForm()) {
         setEnableSubmitButton(true);
+      } else {
+        setEnableSubmitButton(false);
       }
     },
     [firstName, lastName, email, password, confirmPassword]

```